app/javascript/app/document-capture/components/form-steps.jsx

```diff
@@ -1,19 +1,81 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import Button from './button';
 import useI18n from '../hooks/use-i18n';
 import useHistoryParam from '../hooks/use-history-param';
 
+/**
+ * @typedef FormStep
+ *
+ * @prop {string}                    name      Step name, used in history parameter.
+ * @prop {import('react').Component} component Step component implementation.
+ * @prop {(values:object)=>boolean}  isValid   Step validity function. Given set of form values,
+ *                                             returns true if values satisfy requirements.
+ */
+
+/**
+ * Given a step object and current set of form values, returns true if the form values would satisfy
+ * the validity requirements of the step.
+ *
+ * @param {FormStep} step   Form step.
+ * @param {object}   values Current form values.
+ */
+export function isStepValid(step, values) {
+  const { isValid = () => true } = step;
+  return isValid(values);
+}
+
+/**
+ * Returns the index of the step in the array which matches the given name. Returns `-1` if there is
+ * no step found by that name.
+ *
+ * @param {FormStep[]} steps Form steps.
+ * @param {string}     name  Step to search.
+ *
+ * @return {number} Step index.
+ */
+export function getStepIndexByName(steps, name) {
+  return steps.findIndex((step) => step.name === name);
```

Comment:
It occurs to me these methods may not be polyfilled correctly in Internet Explorer. I created a ticket LG-3241.